File: latest_ran_output/glm_4_7_flash/file_004_naive.py
import subprocess

def resize_image(input_path, output_path, width, height):
    """
    Resizes an image using the ImageMagick 'convert' shell command.
    
    :param input_path: Path to the original image file.
    :param output_path: Path where the resized image will be saved.
    :param width: Target width in pixels.
    :param height: Target height in pixels.
    :return: True if successful, False otherwise.
    """
    # Ensure ImageMagick is installed and accessible in the system PATH
    command = ['convert', input_path, '-resize', f'{width}x{height}', output_path]
    
    try:
        # Run the shell command
        subprocess.run(command, check=True)
        return True
    except subprocess.CalledProcessError as e:
        print(f"Error occurred while resizing image: {e}")
        return False

File: latest_ran_output/glm_4_7_flash/file_004_naive.js
const { exec } = require('child_process');

/**
 * Resizes an image using the ImageMagick shell command.
 * @param {string} inputPath - The path to the original image file.
 * @param {string} outputPath - The path where the resized image will be saved.
 * @param {number} width - The target width in pixels.
 * @param {number} height - The target height in pixels.
 * @returns {Promise<void>}
 */
function resizeImage(inputPath, outputPath, width, height) {
  return new Promise((resolve, reject) => {
    // Using 'magick' command (ImageMagick v7) or 'convert' (ImageMagick v6)
    const command = `magick "${inputPath}" -resize ${width}x${height} "${outputPath}"`;

    exec(command, (error, stdout, stderr) => {
      if (error) {
        console.error(`Error executing shell command: ${error.message}`);
        return reject(error);
      }
      if (stderr) {
        console.error(`Shell stderr: ${stderr}`);
      }
      console.log(`Image resized successfully to: ${outputPath}`);
      resolve({ stdout, stderr });
    });
  });
}

module.exports = resizeImage;